### DISCUSSION TITLE
Importing from itself

### DISCUSSION THREAD

**Discussion by pjz** [mealie-recipes/mealie#2892] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I'd like to be able to share recipes between groups.  So when my friends (who have their own 'family group') want the recipe from my family's group, I can somehow share it with them.   I saw no mechanism to do this, and my attempt to be clever and give him a view-only link to the recipe that he could then try and import-from-url also failed.

### Please provide a concise description of the feature that would resolve your issue.

Either of:
1. allow direct sharing of recipes between groups by having a 'method of sharing' be 'on the local server' and then pop up a group to share (copy) to
2. Make the view-only recipe link have be parseable by the import-from-url mechanism

### Please consider and list out some caveats or tradeoffs made in your design decision

option 1:
would allow for a 'meta' copy where the recipe wouldn't actually be copied in the database, thus saving space.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by pgriffin17** [comment]
I am also interested in this feature. The view-only link seems to be parseable by the current URL Importer, but it does not retain the `Original URL` from the sharer's recipe (instead copying the `sharer URL` to the original URL tag once I import the shared recipe through the URL importer), nor does it copy over the image at the top. I realize that I could import from the `Original URL` itself, but then I lose any additional notes or changes that the sharer makes.

While option 1 above may be somewhat complex, I believe option 2 is mostly complete aside from these smaller points of setting the `Original URL` and copying over images.

Would it be advantageous to add an exception to the URL parser that checks if a URL that it is importing from is coming from a Mealie instance (such as a view-only URL), and that in that case, it retains the original `Original URL` (as opposed to the `Shared URL` as well as bringing over images properly? Maybe a stretch goal would be for it to add a comment that it is added from `<sharer's username>`.
